docs/tofu.logging.recipes.zio2.md

```diff
@@ -0,0 +1,213 @@
+  },
+  "zAnnotations": {
+    "foo": "bar"
+  }
+}
+```
+
+* __logger_name__ is parsed from `zio.Trace` which contains the location where log method is called
+* all `zio.LogSpan` are collected in the json object named __zSpans__
+* all `zio.LogAnnotation` are collected in the json object named __zAnnotations__ (to avoid conflicts with Tofu
+  annotations)
+
+### ZLogAnnotation
+
+A specialized version of [LogAnnotation](./tofu.logging.annotation.md) allows you to add a context via ZIO aspects:
+
+```scala
+import tofu.logging.zlogs.ZLogAnnotation._
+
+val httpCode: ZLogAnnotation[Int] = make("httpCode")
+
+override def run = {
+  program @@ httpCode(204) @@ loggerName("MyLogger")
+}.provide(
+  Runtime.removeDefaultLoggers,
+  TofuZLogger.addToRuntime
+)
+```
+
+will produce:
+
+```json
+{
+  "level": "INFO",
+  "logger_name": "MyLogger",
+  "message": "Hello, ZIO logging!",
+  "httpCode": 204
+}
+```
+
+You can change the logger name via `ZLogAnnotation.loggerName`.
+
+`ZLogAnnotation.make[A]` implicitly requires a `Loggable[A]` instance, see more
+in [Loggable](./tofu.logging.loggable.md) section.
+
+### TofuDefaultContext
+
+Using this service you can look up an element from the context added via `ZLogAnnotation`:
+
+```scala
+val httpCode: ZLogAnnotation[Int] = make("httpCode")
+
+val program = {
+  for {
+    maybeCode <- ZIO.serviceWithZIO[TofuDefaultContext](_.getValue(httpCode)) // Some(204)
+    //...
+  } yield ()
+} @@ httpCode(204)
+```
+
+## ZIO implementation of Tofu Logging
+
+If you want more flexible Tofu Logging, `tofu-zio2-logging` provides some useful stuff:
+
+* `ULogging` - is a type alias for `Logging[UIO]`, logging methods of this logger look
+  like `def info(message: String, values: LoggedValue*): UIO[Unit]`
+
+* Logger factory type aliases:
+    - `ZLogging.Make` - is a type alias for `Logs[Id, UIO]`, produces plain instances of `ULogging`.
+    - `ULogs` - is a type alias for `Logs[UIO, UIO]`, produces instances of `ULogging` inside `UIO` effect.
+
+* `ZLogging.Make` and `ZLogs` objects provide corresponding factory instances
+    - `layerPlain` creates layer contains simple implementation of `ZLogging.Make` (or `ULogs`)
+    - `layerPlainWithContext` creates layer with an implementation of `ZLogging.Make` (or `ULogs`) producing loggers
+      which add the context to your logs from the `ContextProvider`.
+      This one is supposed to be provided at the app creation point via ZLayer-s.
+    - `layerContextual[R: Loggable]` makes a factory `ZMake[R]` (or `ZLogs[R]`) of contextual `ZLogging[R]` retrieving a
+      context from
+      the ZIO environment of the logging methods. This legacy approach is contrary to
+      ZIO [Service Pattern](https://zio.dev/reference/service-pattern/), so we won't cover it here.
+
+### ContextProvider
+
+```scala
+trait ContextProvider {
+  def getCtx: UIO[LoggedValue]
+}
+```
+
+Is required by `layerPlainWithContext` factory, every logger will log the provided `LoggedValue`.
```

Comment:
I think we need a bit more motivation/description here. At first glance it seems like a static context provider or something.